src/Components/Web.JS/src/Boot.Server.ts

```diff
@@ -27,16 +28,23 @@ async function boot(userOptions?: Partial<BlazorOptions>): Promise<void> {
   options.reconnectionHandler = options.reconnectionHandler || window['Blazor'].defaultReconnectionHandler;
   logger.log(LogLevel.Information, 'Starting up blazor server-side application.');
 
-  const initialConnection = await initializeConnection(options, logger);
-  const circuit = await startCircuit(initialConnection);
+  const components = discoverComponents(document);
+  const circuit = new CircuitDescriptor(components);
+
+  const initialConnection = await initializeConnection(options, logger, circuit);
```

Comment:
It seems a little weird to me that we pass in the circuit here so that we can call methods on it. I don't have much of a better idea than defining a *callbacks* object to pass in, so I guess I'll leave it up to. 

This factoring led me to believe that the circuit had some role in opening the connection, and it doesn't.

Reply:
i tried the callback approach before and it was cumbersome. This is a pragmatic balance I think.